.circleci/config.yml

```diff
@@ -0,0 +1,42 @@
+version: 2.1
+orbs:
+  codecov: codecov/codecov@1.0.2
+jobs:
+  build:
+    docker:
+      - image: circleci/golang:1.12
+        environment:
+          GO111MODULE: "on"
+    working_directory: /go/src/github.com/regen-network/cosmos-upgrade-manager
+    environment:
+      TEST_RESULTS: /tmp/test-results
+      CODECOV_TOKEN: "b9cd3f1e-5d65-4554-8eb9-292ef46298dc"
+    steps:
+      - checkout
+      - run: mkdir -p $TEST_RESULTS # create the test results directory
+      - restore_cache:
+          keys:
+            - go-mod-v4-{{ checksum "go.sum" }}
+      - run:
+          name: test
+          command: |
+            PACKAGE_NAMES=$(go list ./... | circleci tests split --split-by=timings --timings-type=classname)
```

Comment:
I would prefer `go test .` here.
If you want coverage, please add a Makefile command, then refer to it in this file.

I don't like such code (that is very hard to test locally) in the CI scripts, they should just call git and make, etc.
If you want to add `make test` and `make cover` (this one), that is a great addition.

Reply:
I agree that Makefile in general is preferable. In this case I just copied CircleCI's example to get it integrated well. Are you suggesting I use `go test .` instead of the `gotestsum` which produces XML outputs for Circle CI? I guess there's a question of how much we try to integrate the script with the CI vs making it generalized

Reply:
Well, I usually test without coverage, unless we need it.
And then, usually a (fast) test call, then a slower (coverage) call when it passes.

Failure output tends to be nicer with the normal test options.

Reply:
I can extend this PR a bit...

Reply:
Fixed